src/shared/os-util.c

```diff
@@ -117,3 +117,26 @@ int load_os_release_pairs(const char *root, char ***ret) {
 
         return load_env_file_pairs(f, p, ret);
 }
+
+int load_os_release_pairs_with_prefix(const char *root, const char *prefix, char ***ret) {
+        _cleanup_strv_free_ char **os_release_pairs = NULL;
+        char **p, **q;
+        int r;
+
+        r = load_os_release_pairs(root, &os_release_pairs);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH_PAIR(p, q, os_release_pairs) {
+                char *line;
+
+                line = strjoin(prefix, *p, "=", *q);
+                if (!line)
+                        return -ENOMEM;
+                r = strv_consume(ret, line);
```

Comment:
Per our coding style we have the rule: never clobber return parameters on failure and always initialize them on success. In this case you'll leave a half-built-up strv in the return parameter if strjoin() fails eventually.

Reply:
Fixed, thanks.